src/ucm/util/reloc.h

```diff
@@ -46,4 +47,74 @@ ucs_status_t ucm_reloc_modify(ucm_reloc_patch_t* patch);
 void* ucm_reloc_get_orig(const char *symbol, void *replacement);
 
 
+extern pthread_mutex_t ucm_reloc_get_orig_lock;
```

Comment:
need to create new file: replace,h, replace.c
"reloc" is handling changing program relocation tables
these are macros to generate replacement function

Reply:
moved headers to new file utils/replace.h. 
Also unified {cuda,mmap}/replace.c to util/replace.c.  is this ok?